### ISSUE TITLE
css border shorthand incorrectly minified in production

### ISSUE THREAD

**Issue by stefanprobst** [vercel/next.js#35219]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Operating System:
      Platform: linux
      Arch: x64
      Version: #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: 6.30.0
    Relevant packages:
      next: 12.1.1-canary.9
      react: 17.0.2
      react-dom: 17.0.2

### Describe the Bug

currently, the css minifier (i think this is the compiled-in cssnano-simple) produces invalid css from the following:

```css
h1 {
  --border-width: 8px 2px;

  border-width: var(--border-width);
  border-style: solid;
  border-color: hotpink;
}
```

results in:

```css
h1{--border-width:8px 2px;border:var(--border-width) solid hotpink}
```

which is invalid.

note that this is currently also handled incorrectly in upstream `cssnano` with the default preset, but works correctly with the "lite" preset. cf. https://github.com/cssnano/cssnano/issues/1354

### Expected Behavior

don't collapse border properties into shorthand when custom properties are encountered

### To Reproduce

- checkout https://github.com/stefanprobst/next-issue-css-shorthand and compare the results of `yarn dev` vs `yarn build && yarn start`

**Comment by wwwdepot** [comment]
It's an optimization enabled by cssnano, See `mergeLonghand` https://cssnano.co/docs/what-are-optimisations/.

**Comment by balazsorban44** [comment]
Next.js is using a custom variant of the `cssnano-preset-default` preset, source here: https://github.com/Timer/cssnano-preset-simple

Adding `mergeLonghand: false` to that and re-compile it for Next.js might fix the issue, but it looks like `cssnano` acknowledges this as a bug and we should probably wait for a proper fix on their side: https://github.com/cssnano/cssnano/issues/675 :thinking:

**Comment by stefanprobst** [comment]
there is now a fix for this issue in upstream `cssnano` v5.1.3
